src/main/kotlin/calculator/StringSeparator.kt

```diff
@@ -0,0 +1,24 @@
+package calculator
+
+object StringSeparator {
+
+    private const val DEFAULT_DELIMITER = ",|:"
+    private val CUSTOM_REGEX = Regex("//(.)\n(.*)")
+    private val DEFAULT_NUMBER = PositiveNumbers(listOf(PositiveNumber(0)))
+
+    fun separate(inputString: String?): PositiveNumbers {
+        if (inputString.isNullOrEmpty()) {
+            return DEFAULT_NUMBER
+        }
+
+        return CUSTOM_REGEX.find(inputString)
+            ?.let { toPositiveNumbers(inputString = it.groupValues[2], delimiter = it.groupValues[1]) }
+            ?: toPositiveNumbers(inputString)
+    }
+
+    private fun toPositiveNumbers(inputString: String, delimiter: String = DEFAULT_DELIMITER): PositiveNumbers = inputString.split(delimiter.toRegex())
+        .map {
+            it.toIntOrNull()?.let { intValue -> PositiveNumber(intValue) }
+                ?: throw IllegalArgumentException("숫자가 아닌경우 변환할수 없습니다.")
+        }.run { PositiveNumbers(this) }
```

Comment:
```suggestion
        }.let(::PositiveNumbers)
```
명시적인 리시버를 활용하는 경우에는 run 보다는 let을 사용하는 것이 좋아요.
그리고 파라미터의 타입 및 개수가 일치하다면, 메서드 참조를 활용할 수 있어요.